app/main.py

```diff
@@ -1 +1,27 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100, hidden: bool = False):
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self):
+        return f"{{Name: {self.name}," \
+               f" Health: {self.health}," \
+               f" Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self):
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(other):
+        if isinstance(other, Herbivore) and not other.hidden:
+            other.health -= 50
+            if not other.health:
```

Comment:
What if after the bite Herbivore health become -10?

Reply:
yes, i'll fix  it